src/yachtzee/lib.ts

```diff
@@ -0,0 +1,102 @@
+import { cliExecute, Effect, Item, useFamiliar } from "kolmafia";
+import {
+  $effect,
+  $familiar,
+  $item,
+  $items,
+  $location,
+  adventureMacroAuto,
+  get,
+  getActiveSongs,
+  getModifier,
+  have,
+  Mood,
+  Requirement,
+  set,
+  sum,
+  tryFindFreeRun,
+} from "libram";
+import { withStash } from "../clan";
+import { Macro } from "../combat";
+import { EmbezzlerFight, embezzlerSources } from "../embezzler";
+import { freeFightFamiliar } from "../familiar";
+import { globalOptions, ltbRun, realmAvailable } from "../lib";
+import { freeFightOutfit } from "../outfit";
+
+const ignoredSources = [
+  "Orb Prediction",
+  "Pillkeeper Semirare",
+  "Lucky!",
+  "11-leaf clover (untapped potential)",
+];
+export const expectedEmbezzlers = sum(
+  embezzlerSources.filter((source: EmbezzlerFight) => !ignoredSources.includes(source.name)),
+  (source: EmbezzlerFight) => source.potential()
+);
+
+export function pyecAvailable(): boolean {
+  if (get("_PYECAvailable") === "") {
+    set(
+      "_PYECAvailable",
+      get("expressCardUsed")
+        ? false
+        : have($item`Platinum Yendorian Express Card`)
+        ? true
+        : withStash($items`Platinum Yendorian Express Card`, () => {
+            return have($item`Platinum Yendorian Express Card`);
+          })
+    );
+  }
+  return get("_PYECAvailable", false);
+}
+
+export function shrugIrrelevantSongs(): void {
+  for (const song of getActiveSongs()) {
+    const slot = Mood.defaultOptions.songSlots.find((slot) => slot.includes(song));
+    if (
+      !slot &&
+      song !== $effect`Ode to Booze` &&
+      song !== $effect`Polka of Plenty` &&
+      song !== $effect`Chorale of Companionship` &&
+      song !== $effect`The Ballad of Richie Thingfinder`
+    ) {
+      cliExecute(`shrug ${song}`);
+    }
+  }
+  // Shrug default Mood songs
+  cliExecute("shrug ur-kel");
+  cliExecute("shrug phat loot");
+}
+
+export const freeNCs = (): number =>
+  (have($item`Clara's bell`) && !globalOptions.clarasBellClaimed ? 1 : 0) +
+  (have($item`Jurassic Parka`) ? 5 - get("_spikolodonSpikeUses") : 0);
+
+export function yachtzeeBuffValue(obj: Item | Effect): number {
+  return (2000 * (getModifier("Meat Drop", obj) + getModifier("Familiar Weight", obj) * 2.5)) / 100;
+}
+
+export function useSpikolodonSpikes(): void {
+  if (get("_spikolodonSpikeUses") >= 5) return;
+  const run = tryFindFreeRun() ?? ltbRun();
```

Comment:
Are we saving free runs for this? It might be prudent to head into `fights.ts` to disable all the FreeRunFights with jellyfish if we are chaining and have spikolodon spikes available.

Reply:
I don't know if this is a `when`, given that we have a lot of potential freerun sources. ltbRun technically also includes tennis balls and champagne poppers.

Right now, we are not actively saving freeruns for this; the issue with cutting out the jellyfish runs is that those will just get used fishing for hipsters and whatall.

What we may be able to do is push freeRunFights to after we do yachtzee, when yachtzeeing. We'll have to think about if there's a compelling reason not to do that.

Reply:
1 reason I can think of not doing it is yachtzeeing will (out of neccessity) either wholesale use up (or at least partially eat into) fam weight buffs, which overall would reduce the amount of potential familiar free runs we can do. So if we were to push freeRunFights to post yachtzee, we may want to split it out with "free fam runs" pre yachtzee, and others post